src/checks.ts

```diff
@@ -99,6 +99,14 @@ export const Checks = {
       /** @todo `Checks.isFunction` が実装されたらそれを使う */
       getObjectTypeName((value as MaybeThenable).then) === '[object Function]'
     )
+  },
+
+  /**
+   * 値が `Map` か否かを返す
+   * @param value
+   */
+  isMap<K, V>(value: unknown): value is Map<K, V> {
```

Comment:
この関数で判定できるのは Map インスタンスであるかどうか。
なので型的には以下のほうが正確。

```suggestion
  isMap(value: unknown): value is Map<unknown, unknown> {
```

便利さをとって型安全を捨てるか、便利さを捨てて型安全性を取るかみたいな…。
ご意見求ム。

Reply:
`isMap(value: unknown): value is Map<unknown, unknown> {`

とする場合は `isPromise` (`isPromiseLike`) も

```ts
isPromise(value: unknown): value is Promise<unknown> {
```

にする。

Reply:
@isoden ジェネリクスやってるとキリが無いので `unknown` で良い気がしてきました。 なお、 `isArray()` も修正必要です。

Reply:
#43 で対応しました。

Reply:
@haribote `isMap` からも generics 外しました。改めてレビューお願いします :pray:

Promise 系は別でPR投げます。